src/main/java/de/zalando/tip/zalenium/proxy/CloudTestingRemoteProxy.java

```diff
@@ -0,0 +1,148 @@
+package de.zalando.tip.zalenium.proxy;
+
+/*
+    Almost all concepts and ideas for common to the implementation that are inherited by the Sauce Labs and
+    BrowserStack proxy are taken from the open source project seen here: https://github.com/rossrowe/sauce-grid-plugin
+ */
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import de.zalando.tip.zalenium.util.CommonProxyUtilities;
+import de.zalando.tip.zalenium.util.GoogleAnalyticsApi;
+import de.zalando.tip.zalenium.util.ZaleniumCapabilityMatcher;
+import org.openqa.grid.common.RegistrationRequest;
+import org.openqa.grid.internal.Registry;
+import org.openqa.grid.internal.TestSession;
+import org.openqa.grid.internal.utils.CapabilityMatcher;
+import org.openqa.grid.selenium.proxy.DefaultRemoteProxy;
+import org.openqa.grid.web.servlet.handler.RequestType;
+import org.openqa.grid.web.servlet.handler.WebDriverRequest;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class CloudTestingRemoteProxy extends DefaultRemoteProxy {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Override this superclass' "equals" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2160)